src/Vector.cpp

```diff
@@ -0,0 +1,335 @@
+	++m_size;	
+}
+
+/**
+ * erase the element at the given position
+ * @param pos the position to erase
+ **/
+void Vector::erase(size_t pos)
+{
+	//erase the element at the given pos
+	if (pos >= m_size) {
+		throw std::out_of_range("Index is out of range");
+	}
+	for (size_t i = pos; i < m_size - 1; ++i) {
+		m_vector[i] = m_vector[i + 1];
+	}
+	--m_size;
+}
+
+// ==================== operators overloading ====================
+
+/**
+ * operator= assigns a vector to another vector
+ * @param other the vector to assign
+ * @return the vector after the assignment
+ **/
+Vector& Vector::operator=(const Vector& other)
+{
+	// copy and swap idiom implementation
+	if (this != &other)
+	{
+		Vector tmp(other); // use the copy ctor to create a new temp vector
+		this->swap(tmp); // swap the current vector with the temp vector
+	}
+	return *this; 
+}
+
+/**
+* operator= move a vector to another vector
+* @param other the vector to move
+* @return the vector after moving other to it
+**/
+Vector& Vector::operator=(Vector&& other) noexcept
+{
+	// move and swap idiom implementation
+	if (this != &other)
+	{
+		this->swap(other);
+		other.m_vector = nullptr;
+		other.m_size = other.m_capacity = 0;
+	}
+	return *this;
+}
+
+/**
+ * operator[] regular and const versions
+ * @param index the index of the element to return
+ * @return the element at the given index
+ **/
+int& Vector::operator[](size_t index)
+{
+	if (index >= m_size) {
+		throw std::out_of_range("Index out of range");
+	}
+	return m_vector[index];
+}
+const int& Vector::operator[](size_t index) const
+{
+	if (index >= m_size) {
+		throw std::out_of_range("Index out of range");
+	}
+	return m_vector[index];
+}
+
+/**
+ * operator== checks if two vectors are equal
+ * @param other the vector to compare with
+ * @return true if the vectors are equal, false otherwise
+ **/
+bool Vector::operator==(const Vector& other) const
+{
+	if (m_size == other.m_size && 
+		std::equal(m_vector, m_vector + m_size, other.m_vector)) {
+			return true;
+	}
+	return false;
+}
+
+bool Vector::operator!=(const Vector& other) const
+{
+	return !(*this == other);
+}
+
+/**
+ * operator<=> compares two vectors
+ * @param other the the vector to compare with  
+ * @return the equal less or greater  
+ **/
+std::strong_ordering Vector::operator<=>(const Vector& other) const noexcept
+{
+	if (*this == other) {
+		return std::strong_ordering::equal;
+	} 
+	else if (m_size < other.m_size) {
+		return std::strong_ordering::less;
+	}
+	return std::strong_ordering::greater;
+}
+
+// ==================== iterators ================================
+
+int* Vector::begin() const
+{
+	return m_vector;
+}
+
+int* Vector::end() const
+{
+	return m_vector + m_size;
+}
+
+// ==================== private methods ==========================
+
+/**
+ * check_capacity checks if the vector is full and if so, reserve more memory
+ **/
+void Vector::check_capacity()
+{
+	// if capacity is 0, reserve 1 cell
+	if (m_capacity == 0) {
```

Comment:
This `if` can be spared if you originally set `m_capacity` and the vector size to something larger than 0. 
This will give you a performance boost

Reply:
fixed